internal/prometheus/middleware.go

```diff
@@ -18,3 +20,35 @@ func MetricsMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
 		return next(ctx)
 	}
 }
+
+func StatusMiddleware(subsystem string) func(next echo.HandlerFunc) echo.HandlerFunc {
+	counter := StatusRequestsCounter(subsystem)
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(ctx echo.Context) error {
+
+			// call the next handler to see if
+			// an error occurred, see:
+			// - https://github.com/labstack/echo/issues/1837#issuecomment-816399630
+			// - https://github.com/labstack/echo/discussions/1820#discussioncomment-529428
+			err := next(ctx)
+
+			path := pathLabel(ctx.Path())
+			method := ctx.Request().Method
+			status := ctx.Response().Status
+
+			httpErr := new(echo.HTTPError)
+			if errors.As(err, &httpErr) {
+				status = httpErr.Code
+			}
```

Comment:
Does the code in the echo error sometimes differ from the status one?

Also any reason for using As over Is?

Reply:
So the status code before that check is always `200`. If I used Is over As, it doesn't pick it up and the error code stays `200`.